argilla-server/src/argilla_server/api/handlers/v1/oauth2.py

```
@@ -74,32 +73,22 @@ async def get_access_token(
     provider: OAuth2ClientProvider = Depends(get_provider_by_name_or_raise),
     db: AsyncSession = Depends(get_async_db),
 ) -> Token:
-    try:
-        user_info = UserInfo(await provider.get_user_data(request)).use_claims(provider.claims)
-        user = await User.get_by(db, username=user_info.username)
-        if user is None:
-            try:
-                user_create = UserOAuthCreate(
-                    username=user_info.username,
-                    first_name=user_info.first_name,
-                    role=user_info.role,
-                )
-            except ValidationError as ex:
-                raise AuthenticationError("Could not authenticate user") from ex
-
-            user = await accounts.create_user_with_random_password(
-                db,
-                **user_create.dict(exclude_unset=True),
-                workspaces=[workspace.name for workspace in settings.oauth.allowed_workspaces],
-            )
-            telemetry.track_user_created(user, is_oauth=True)
-
-        elif user.role != user_info.role:
-            raise AuthenticationError("Could not authenticate user")
-
-        return Token(access_token=JWT.create(user_info))
-    except ValueError as e:
-        raise HTTPException(status_code=400, detail=str(e)) from e
-    # TODO: Create exception handler for AuthenticationError
-    except AuthenticationError as e:
-        raise HTTPException(status_code=401, detail=str(e)) from e
+    userinfo = UserInfo(await provider.get_user_data(request)).use_claims(provider.claims)
+
+    if not userinfo.username:
+        raise RuntimeError("OAuth error: Missing username")
```

This is an unexpected error and probably the OAuth config is wrong. It will raise a 500 error

---

then do you think it's ok to don't check for that condition and let it crash on line 81?

---

No, I prefer to raise this error instead of a validation one.